src/util/simplify_expr.cpp

```diff
@@ -1536,8 +1536,13 @@ exprt simplify_exprt::bits2expr(
   else if(type.id()==ID_array)
   {
     const array_typet &array_type=to_array_type(type);
+    const auto &size = array_type.size();
 
-    const std::size_t n_el = numeric_cast_v<std::size_t>(array_type.size());
+    if(!size.is_constant())
```

Comment:
Change in behaviour: returning empty array vs. nil?

Reply:
That one seems like a bugfix!

Reply:
This is a change in behaviour - could you add a unit test checking this and ideally put it into a separate commit in case it breaks anything (this would be a big commit to comb through!)

Presumably this change came about, because in a regression test the size here is non-const, perhaps the test could be tightened so it would have failed before too?